src/Executables/Game/World/Map.cs

```diff
@@ -279,22 +279,47 @@ private MonsterEntity SpawnMonster(uint id, SpawnPoint spawnPoint)
             var baseY = spawnPoint.Y;
             if (spawnPoint.RangeX != 0)
             {
-                baseX += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeY);
+                baseX += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeX);
             }
 
             if (spawnPoint.RangeY != 0)
             {
-                baseY += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeY);
+                baseY += RandomNumberGenerator.GetInt32(-spawnPoint.RangeY, spawnPoint.RangeY);
             }
 
-            var monster = new MonsterEntity(_monsterManager, _dropProvider, _animationManager, this, _logger, _itemManager,
+            var monster = new MonsterEntity(_monsterManager, _dropProvider, _animationManager, this, _logger,
+                _itemManager,
                 id,
-                (int) (PositionX + (baseX + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) *
-                    SPAWN_POSITION_MULTIPLIER),
-                (int) (PositionY + (baseY + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) *
-                    SPAWN_POSITION_MULTIPLIER),
-                RandomNumberGenerator.GetInt32(0, 360));
-            _world.SpawnEntity(monster);
+                0,
+                0
+            );
+
+            if (((uint)EAiFlags.NoMove & monster.Proto.AiFlag) == (uint)EAiFlags.NoMove) // no_move
```

Comment:
You can use Enums.NET's `HasAnyFlag` which is as performant as this but better to read :)
The property has to be an enum for that obviously

Reply:
Would this work? My instincts says it wouldn't, because some mobs have a combination of flags. Wouldn't casting `monster.Proto.AiFlag` to the `EAiFlags` enum throw an exception if the value isn't defined from the enum?

```cs
if (((EAiFlags)monster.Proto.AiFlag).HasAnyFlags(EAiFlags.NoMove)) {
    //...
}
```

Reply:
No it wouldn't throw an exception because I are not actually casting (only a primitive cast of the underlying type - byte to byte would result in nocast). If the value is not defined on the enum you couldn't check for a bit via named enum flags but that's about it.